src/main/scala/mesosphere/marathon/core/task/jobs/impl/OverdueInstancesActor.scala

```diff
@@ -99,7 +99,7 @@ private[jobs] object OverdueInstancesActor {
 
     private[this] def overdueReservations(now: Timestamp, instances: Seq[Instance]): Seq[Instance] = {
       instances.filter { instance =>
-        instance.isScheduled && instance.reservation.exists(_.state.timeout.exists(_.deadline <= now))
+        (instance.isScheduled || instance.state.goal == Goal.Decommissioned) && instance.reservation.exists(_.state.timeout.exists(_.deadline <= now))
```

Comment:
So. This works for the case we've encountered on soak114. 
The big question is: Do we actually need the first part of this check. Is there any case, condition or goal, where - when the requested reservation timed out, we do *not* want to force expunge that instance?

Reply:
Really nice find! 👏